### ISSUE TITLE
how to get the visualization of the segmentation results

### ISSUE THREAD

**Issue by ABFHJJGGGV** [matteo-bastico/MI-Seg#2]
HI,Thank you for your research and share.
May I ask how to get the visualization of the segmentation results in the paper.
Thank you, very much.

**Comment by matteo-bastico** [comment]
Hello, you can use the following code to plot all the slices of two 3D images:

```
import nibabel as nib
import matplotlib.pyplot as plt

# Define colormap
cmap = matplotlib.colors.LinearSegmentedColormap.from_list(
    "", 
    ["black","m","b", "darkviolet", "lightblue", "g", "orange", "red"],
)

# Load 3D volumes
gt = nib.load('path/to/gt.nii.gz')
gt = np.array(gt.dataobj)
pred = nib.load('path/to/pred.nii.gz')
pred = np.array(pred.dataobj)

# Plot all the slices 
for slice in range(0, gt.shape[2]):
    plt.figure(figsize=(10, 5))
    plt.subplot(1, 2, 1)
    plt.imshow(gt[:, :, slice], cmap=cmap, vmax=7, vmin=0, interpolation='none')
    plt.axis('off')
    plt.subplot(1, 2, 2)
    plt.imshow(pred[:, :, slice], cmap=cmap, vmax=7, vmin=0, interpolation='none')
    plt.axis('off')
    plt.show()
```

If you want to compare more results, you can just load more volumes and add columns in the subplot. Note that the segmentation labels should be in the interval [0,7], as we pre-process in the  `dataset/MM-WHS/load_data.ipynb` notebook [here](https://github.com/matteo-bastico/MI-Seg/blob/master/dataset/MM-WHS/load_data.ipynb).

**Comment by ABFHJJGGGV** [comment]
Thank you for your reply. I want to know how to get gt.nii.gz and pred.nii.gz files. Because I do not know how to save the results of the segmentation output as a file.I am a beginner of scientific research, thank you for your patient reply. Good luck in scientific research

**Comment by matteo-bastico** [comment]
Sorry for the delayed response on this issue. 🙇‍♂️ You can download the gt segmentation images from the [MM-WHS dataset page](https://zmiclab.github.io/zxh/0/mmwhs/). All the files ending with `_label.nii.gz` are the gt of the CT or MR image with the same id. 

> Note that for the test sets the gt images are encrypted and therefore cannot be directly visualized. 

The model predictions can be generated using  `predict_whs.py` script. Here you have to provide the trained model weights with the `--checkpoint` argument (you can train a model with your data or use our trained model weights which will be released soon) and the list of images to segment, in a json file like the provided examples in the `dataset/MM-WHS folder`, with  `--json_list` .

**Comment by matteo-bastico** [comment]
Pre-trained model weights added on commit 8ff7c7d937dc41516347b67a631862302172cdcf
